mypy.ini

```diff
@@ -10,5 +10,9 @@ enable_error_code = ignore-without-code
 # Support namespace packages per https://github.com/python/mypy/issues/14057
 explicit_package_bases = True
 
-# Disable overload-overlap due to many false-positives
-disable_error_code = overload-overlap
+disable_error_code =
+	# Disabled due to many false-positives
+	overload-overlap,
```

Comment:
> many false-positives

for instance...?

Reply:
Often triggered by overloads containing type-vars or default values.

Currently in setuptools: `setuptools\monkey.py:43: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [overload-overlap]`

There was consideration to disable it in typeshed itself. But so far it looks like it'll be kept simply to help inform how it can be improved: https://github.com/python/typeshed/issues/12178
![image](https://github.com/user-attachments/assets/4468d847-82af-4bb3-9331-6ac79a8ea8ae)

Many related open mypy issues: https://github.com/python/mypy/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+overload-overlap+label%3Atopic-overloads+

But to manually filter them down a bit more:
- https://github.com/python/mypy/issues/6580
- https://github.com/python/mypy/issues/6623
- https://github.com/python/mypy/issues/10157
- https://github.com/python/mypy/issues/10143
- https://github.com/python/mypy/issues/13611
- https://github.com/python/mypy/issues/14770
- https://github.com/python/mypy/issues/11165

Reply:
Note that this PR doesn't change that setting, but merely moves it to its own line for easier diffs in the future. Since I'll not be accepting this change, I'll push a separate change with just the reformatting.